src/test/java/de/tum/in/www1/artemis/assessment/AssessmentComplaintIntegrationTest.java

```diff
@@ -602,17 +600,21 @@ void getSubmittedComplaints_byComplaintType() throws Exception {
         complaintUtilService.addComplaints(TEST_PREFIX + "student1", modelingAssessment.getParticipation(), 1, ComplaintType.COMPLAINT);
         complaintUtilService.addComplaints(TEST_PREFIX + "student1", modelingAssessment.getParticipation(), 2, ComplaintType.MORE_FEEDBACK);
 
-        String exercisesUrl = "/api/exercises/" + modelingExercise.getId() + "/complaints";
-        String coursesUrl = "/api/courses/" + modelingExercise.getCourseViaExerciseGroupOrCourseMember().getId() + "/complaints";
-        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
-        params.add("complaintType", ComplaintType.COMPLAINT.toString());
-        List<Complaint> complaintsByCourse = request.getList(coursesUrl, HttpStatus.OK, Complaint.class, params);
-        List<Complaint> complaintsByExercise = request.getList(exercisesUrl, HttpStatus.OK, Complaint.class, params);
+        String complaintsUrl = "/api/complaints";
+        LinkedMultiValueMap<String, String> paramsExercise = new LinkedMultiValueMap<>();
+        LinkedMultiValueMap<String, String> paramsCourse = new LinkedMultiValueMap<>();
+        paramsExercise.add("complaintType", ComplaintType.COMPLAINT.toString());
+        paramsExercise.add("exerciseId", modelingExercise.getId().toString());
+        paramsCourse.add("complaintType", ComplaintType.COMPLAINT.toString());
+        paramsCourse.add("courseId", modelingExercise.getCourseViaExerciseGroupOrCourseMember().getId().toString());
+        List<Complaint> complaintsByCourse = request.getList(complaintsUrl, HttpStatus.OK, Complaint.class, paramsCourse);
+        List<Complaint> complaintsByExercise = request.getList(complaintsUrl, HttpStatus.OK, Complaint.class, paramsExercise);
         assertThat(complaintsByExercise).hasSameSizeAs(complaintsByCourse).hasSize(1);
```

Comment:
Could you please add a check for `complaintsByCourse` when you're at it? Otherwise, it doesn't make much sense to collect the list here.